engine/src/main/java/org/hibernate/validator/internal/engine/ValidationContext.java

```diff
@@ -292,10 +292,13 @@ public void addConstraintFailures(Set<ConstraintViolation<T>> failingConstraintV
 		);
 		// at this point we make a copy of the path to avoid side effects
 		Path path = PathImpl.createCopy( constraintViolationCreationContext.getPath() );
-
+		//same for expression variables
+		Map<String, Object> expressionVariables =
```

Comment:
No, no problem here. It's not supposed to be modified afterwards, immutability is better IMO, especially when it's so easy to achieve. Plus, I'm not omniscent :p